apps/gdalalg_raster_calc.cpp

```diff
@@ -0,0 +1,619 @@
+
+                CPLXMLNode *sourceFilename =
+                    CPLCreateXMLNode(source, CXT_Element, "SourceFilename");
+                CPLAddXMLAttributeAndValue(sourceFilename, "relativeToVRT",
+                                           "0");
+                CPLCreateXMLNode(sourceFilename, CXT_Text, dsn.c_str());
+
+                CPLXMLNode *sourceBand =
+                    CPLCreateXMLNode(source, CXT_Element, "SourceBand");
+                CPLCreateXMLNode(sourceBand, CXT_Text,
+                                 std::to_string(nInBand).c_str());
+
+                // TODO add <SourceProperties> ?
+
+                CPLXMLNode *srcRect =
+                    CPLCreateXMLNode(source, CXT_Element, "SrcRect");
+                CPLAddXMLAttributeAndValue(srcRect, "xOff", "0");
+                CPLAddXMLAttributeAndValue(srcRect, "yOff", "0");
+                CPLAddXMLAttributeAndValue(srcRect, "xSize",
+                                           std::to_string(props.nX).c_str());
+                CPLAddXMLAttributeAndValue(srcRect, "ySize",
+                                           std::to_string(props.nY).c_str());
+
+                CPLXMLNode *dstRect =
+                    CPLCreateXMLNode(source, CXT_Element, "DstRect");
+                CPLAddXMLAttributeAndValue(dstRect, "xOff", "0");
+                CPLAddXMLAttributeAndValue(dstRect, "yOff", "0");
+                CPLAddXMLAttributeAndValue(dstRect, "xSize",
+                                           std::to_string(nXOut).c_str());
+                CPLAddXMLAttributeAndValue(dstRect, "ySize",
+                                           std::to_string(nYOut).c_str());
+            }
+        }
+
+        // Add the expression as a last step, because we may modify the
+        // expression as we iterate through the bands.
+        CPLAddXMLAttributeAndValue(arguments, "expression",
+                                   bandExpression.c_str());
+        CPLAddXMLAttributeAndValue(arguments, "dialect", "muparser");
+    }
+
+    return true;
+}
+
+static bool ParseSourceDescriptors(const std::vector<std::string> &inputs,
+                                   std::map<std::string, std::string> &datasets,
+                                   std::string &firstSourceName)
+{
+    bool isFirst = true;
+
+    for (const auto &input : inputs)
+    {
+        std::string name = "";
+
+        auto pos = input.find('=');
+        if (pos == std::string::npos)
+        {
+            if (inputs.size() > 1)
+            {
+                CPLError(CE_Failure, CPLE_AppDefined,
+                         "Inputs must be named when more than one input is "
+                         "provided.");
+                return false;
+            }
+            name = "X";
+        }
+        else
+        {
+            name = input.substr(0, pos);
+        }
+
+        std::string dsn = input.substr(pos + 1, input.size());
+        datasets[name] = dsn;
```

Comment:
To please Coverity Scan...
```suggestion
        datasets[name] = std::move(dsn);
```